optimum/onnxruntime/utils.py

```diff
@@ -218,6 +218,28 @@ def validate_provider_availability(provider: str):
         )
 
 
+def possible_io_binding_for_provider(providers: List[str]) -> bool:
+    """
+    Gets the PyTorch device (CPU/CUDA) associated with an ONNX Runtime provider.
+    """
+    io_binding_options = {
+        "TensorrtExecutionProvider": (False,),
+        "CUDAExecutionProvider": (True, False),
+        "CPUExecutionProvider": (False,),
+    }
+
+    if "TensorrtExecutionProvider" in providers:
+        return io_binding_options["TensorrtExecutionProvider"]
+    elif "CUDAExecutionProvider" in providers:
+        return io_binding_options["CUDAExecutionProvider"]
+    elif "CPUExecutionProvider" in providers:
+        return io_binding_options["CPUExecutionProvider"]
+    else:
+        raise KeyError(
+            f"{providers} are not registered execution providers. If you want us to support other ep, please open an issue here: https://github.com/huggingface/optimum/issues."
```

Comment:
could we rather have a logger.warning saying that this EP is not tested, use at your own risk?

Reply:
not sure if we can use something else than CPUExecutionProvider, TensorrtExecutionProvider, CUDAExecutionProvider

Reply:
I would prefer that users let us know if it is needed instead of testing and using/dropping it silently.